### PR TITLE
fix: resolve stuck accounts in process of undelegation

### PR DESCRIPTION
**Pull request by thlorenz**
## 
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved account cleanup to include accounts stuck in intermediate or pending states so eligible accounts are no longer skipped during removal, increasing reliability.

* **Tests**
  * Adjusted integration timing to ensure subscriptions reflect consecutive airdrops and processing delays for consistent balance updates.
  * Increased retry limit in deployment tests to reduce transient failures and improve test stability.

This PR fixes an edge case where accounts in the process of undelegation can get stuck if the validator misses the update of the undelegation completion.
The fix ensures that such accounts are properly cleaned up on restart.

## Details

The issue occurred when accounts were being undelegated but the process didn't complete while the validator was active or we missed the update.

### CODE REVIEW

**Review by bmuddha — APPROVED** [review]
LGTM

File: magicblock-chainlink/src/chainlink/mod.rs [hunk 1]
```diff
@@ -137,7 +137,11 @@ impl<T: ChainRpcClient, U: ChainPubsubClient, V: AccountsBank, C: Cloner>
         let blacklisted_accounts =
             blacklisted_accounts(&self.validator_id, &self.faucet_id);
         let removed = self.accounts_bank.remove_where(|pubkey, account| {
-            !account.delegated() && !blacklisted_accounts.contains(pubkey)
+            (!account.delegated()
+                // This fixes the edge-case of accounts that were in the process of
+                // being undelegated but never completed while the validator was running
+                || account.owner().eq(&dlp::id()))
```
[the remark this answers is not in the document]
Comment (taco-paco) [reply magicblock-chainlink/src/chainlink/mod.rs]: Why is it improper? In context of ER it was undelegated. Everything else is the matter of properly committing it. We shall not bring a network or Intent service related issues into Account state.
In context of ER account it whether delegated or not.

[the remark this answers is not in the document]
Comment (GabrielePicco) [reply magicblock-chainlink/src/chainlink/mod.rs]: I think setting the delegation flag to false is necessary but not sufficient to properly handle all edge cases. We also need additional context to determine when an account is awaiting_undelegation, so we can correctly handle transactions that attempt to use that account while it is in the process of being undelegated versus after it has already been undelegated. In the first case, transactions in the ER should fail. In the second case, the cloning pipeline should "re-fetch" the account
